### ISSUE TITLE
calculating Self consistency loss on depth features

### ISSUE THREAD

**Issue by aneeshc12** [lpiccinelli-eth/UniDepth#51]
Hello authors, good work on the paper and thanks for opensourcing the code!

i'm trying to write my own training loop and I noticed that the SelfCons class seems to work for inputs with the shape (B x H x W) or depth features that have been transposed to be the shape (B x C x h x w). The paper describes self-consistency loss as being calculated from depth features (D|E), but there doesnt seem to be a way to return depth features from the UniDepth_v1 or v2 models. (The v1 model's Decoder class returns depth features).

Are we meant to calculate self-consistency loss from both predicted images? Or do we edit the models to return depth features?

Forgive me if I have misunderstood anything.

**Comment by lpiccinelli-eth** [comment]
Thank you for your question!

We did not expose the depth_features on v2 models, they correspond to line 355 in `decoder.py`, namely the variable `latents_16`, with the next PR we are exposing them again.

The loss accepts the camera information and flipping information to perform the zooming/cropping/translation operations that allow to match the two feature maps and then apply elementwise distance.

**Comment by aneeshc12** [comment]
Thanks for the response!
